ocaml/networkd/lib/network_utils.ml

```diff
@@ -695,13 +695,38 @@ module Ip = struct
     with _ ->
       Result.Error (Fail_to_set_vf_vlan, "Failed to set VF VLAN for: " ^ dev)
 
-  (* We know some NICs do not support config VF Rate, so will explicitly tell
-     XAPI this error*)
+  (** [set_vf_rate dev index rate] Updates the VF rate for a specified network interface card (NIC).
+    Note that this function converts the input rate from kilobits per second (kbps) to
+    megabits per second (Mbps) using division followed by ceil.
+
+    This function accounts for the fact that some NICs may not support VF rate configuration.
+    Any encountered errors during the process are explicitly reported to XAPI.
+
+    @param dev    The network device identifier.
+    @param index  The index of the VF.
+    @param rate   The desired rate in kbps, it will be converted into Mbps using division and ceil.
+
+    @return [Result.Ok] with the updated configuration upon success, or [Result.Error] wrapping [Fail_to_set_vf_rate] with the specific error message upon failure.
+*)
   let set_vf_rate dev index rate =
+    let divide_and_ceil (numerator : int) (denominator : int) : int =
+      let quotient = numerator / denominator in
+      let remainder = numerator mod denominator in
+      if remainder > 0 then
+        quotient + 1
+      else
+        quotient
+    in
     try
-      debug "Setting VF rate for dev: %s, index: %d, rate: %d" dev index rate ;
+      let megabits_per_second = divide_and_ceil rate 1000 in
```

Comment:
I think @robhoes found that this does indeed happen in another place already (that I missed). Except it divides by 1024 and not 1000, so would need to fix up the divisor.

Reply:
See this line in device.ml: 'Network_client.Client.Sriov.{mac= Some mac; vlan; rate= rate_Mbps}'